comp/otelcol/ddprofilingextension/impl/extension_test.go

```diff
@@ -0,0 +1,195 @@
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes/source"
+	ossconfig "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/datadog/config"
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type testComponent struct {
+	*pkgagent.Agent
+}
+
+func (c testComponent) SetOTelAttributeTranslator(attrstrans *attributes.Translator) {
+	c.Agent.OTLPReceiver.SetOTelAttributeTranslator(attrstrans)
+}
+
+func (c testComponent) ReceiveOTLPSpans(ctx context.Context, rspans ptrace.ResourceSpans, httpHeader http.Header) source.Source {
+	return c.Agent.OTLPReceiver.ReceiveResourceSpans(ctx, rspans, httpHeader)
+}
+
+func (c testComponent) SendStatsPayload(p *pb.StatsPayload) {
+	c.Agent.StatsWriter.SendPayload(p)
+}
+
+func (c testComponent) GetHTTPHandler(endpoint string) http.Handler {
+	c.Agent.Receiver.BuildHandlers()
+	if v, ok := c.Agent.Receiver.Handlers[endpoint]; ok {
+		return v
+	}
+	return nil
+}
+
+type hostWithExtensions struct {
+	component.Host
+	exts map[component.ID]component.Component
+}
+
+func newHostWithExtensions(exts map[component.ID]component.Component) component.Host {
+	return &hostWithExtensions{
+		Host: componenttest.NewNopHost(),
+		exts: exts,
+	}
+}
+
+func (h *hostWithExtensions) GetExtensions() map[component.ID]component.Component {
+	return h.exts
+}
+
+func TestNewExtension(t *testing.T) {
+	ext, err := NewExtension(&Config{}, component.BuildInfo{}, testComponent{}, log.NewTemporaryLoggerWithoutInit())
+	assert.NoError(t, err)
+
+	_, ok := ext.(*ddExtension)
+	assert.True(t, ok)
+}
+
+func testServer(t *testing.T, got chan string) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		_, err := io.ReadAll(req.Body)
+		assert.NoError(t, err)
+		assert.Equal(t, "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", req.Header.Get("DD-Api-Key"))
+		got <- req.Header.Get("User-Agent")
+		rw.WriteHeader(http.StatusAccepted)
+	}))
+}
+
+func TestAgentExtension(t *testing.T) {
+	// fake intake
+	got := make(chan string, 1)
+	server := testServer(t, got)
+	defer server.Close()
+
+	// create agent
+	tcfg := config.New()
+	tcfg.ReceiverEnabled = false
+	tcfg.Endpoints[0].APIKey = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
+	tcfg.DecoderTimeout = 10000
+	tcfg.ProfilingProxy = config.ProfilingProxyConfig{DDURL: server.URL}
+	ctx := context.Background()
+	traceagent := pkgagent.NewAgent(ctx, tcfg, telemetry.NewNoopCollector(), &ddgostatsd.NoOpClient{}, gzip.NewComponent())
+
+	// create extension
+	ext, err := NewExtension(&Config{
+		ProfilerOptions: ProfilerOptions{
+			Period: 5,
+		},
+	}, component.BuildInfo{}, testComponent{traceagent}, log.NewTemporaryLoggerWithoutInit())
+	assert.NoError(t, err)
+
+	ext, ok := ext.(*ddExtension)
+	assert.True(t, ok)
```

Comment:
Note: you should probably use `require` instead of `assert` because if `ok == false` then `ext` is `nil` which lead to a panic later in this function. `require` stops the function when the condition is not satisfied.